CMake/Findfoonathan_memory.cmake

```diff
@@ -0,0 +1,35 @@
+
```

Comment:
This looks like a patch,
So we need this empty file to be able to build?

Reply:
CMake looks for this file and, without it, it won't find foonathan. I.e., yes: it's a workaround, but it solves our problems and FastDDS no longer complains.

Reply:
So you are saying lets "lie" to FastDDS and when he say Find(Foonathan) we will say Yes it exist and actually build it later

Reply:
Yep